intelligence/ts/examples/nextjs-web-chats/README.md

```diff
@@ -0,0 +1,83 @@
+# Chat App
+
+A Next.js application (using the App Router, TypeScript, and TailwindCSS) demonstrating three distinct web chat approaches with the [@flwr/flwr](https://www.npmjs.com/package/@flwr/flwr) package.
+
+## Minimum Requirements
+
+- **Node.js** (v14+ recommended)
+- **npm** or **yarn**
+- **Next.js** (using the App Router)
+- **TypeScript**
+- **TailwindCSS**
+- **@flwr/flwr** package installed
+
+## Setup & Installation
+
+```bash
+npm install
+npm run dev
+```
+
+Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+
+## Chat Pages Specificities
+
+### 1. **API Chat (/api-chat)**
+
+- **Approach:** Client component that sends questions to an API route.
+- **Functionality:**
+  - Uses local state to maintain the chat log.
+  - Calls the `/api/chat` endpoint which internally uses shared history (via `lib/chat.ts`) and the @flwr/flwr client.
+  - Renders messages with Markdown formatting.
+- **Use Case:** Ideal for separating client and server logic with API calls.
+
+### 2. **Server-side Chat (/server-side-chat)**
+
+- **Approach:** Server-rendered page using server actions.
+- **Functionality:**
+  - Renders chat history on the server.
+  - Uses a client component (`ChatForm.tsx`) to handle input submission.
+  - The send button is disabled and a loading indicator is displayed (using `useTransition`) while waiting for @flwr/flwr to process the response.
+  - Maintains global conversation history.
+- **Note:** Server components do not support client-side interactivity (like transitions), so the form is split into a client component.
+- **Use Case:** Perfect when you need server-side rendering for SEO or performance while still retaining interactive elements.
+
+### 3. **Client-side Chat (/client-side-chat)**
+
+- **Approach:** Fully client-side component that directly consumes the @flwr/flwr client.
+- **Functionality:**
+  - Uses React hooks (state and effects) to handle messages.
+  - Directly calls the FlowerIntelligence client from `@flwr/flwr` using the shared conversation history from `lib/chat.ts`.
+  - Renders a consistent chat UI with Markdown formatting.
+- **Use Case:** Ideal when you want to handle all interactions on the client for a more dynamic experience.
```

Comment:
@smoroso this gives me CORS error when trying it out on localhost:3000. Any idea why?

Reply:
I don't have that locally but pushing instructions to solve it. Can you try:
- Configure a proxy in your `next.config.ts` to forward API requests to the correct origin:

   ```ts
   import type { NextConfig } from 'next';

   const nextConfig: NextConfig = {
     async rewrites() {
       return [
         {
           source: '/api/:path*',
           destination: 'http://localhost:3000/api/:path*', // Replace with your API server's origin
         },
       ];
     },
   };

   export default nextConfig;
   ```

Reply:
I cannot reproduce but I pushed an instruction in README.md to try in next.config.ts

Reply:
@tanertopal I wasn't able to reproduce the CORS error either